tests/ui/loops/dont-suggest-break-thru-item.rs

```diff
@@ -0,0 +1,26 @@
+fn a() {
+    loop {
+        let closure = || {
+            if true {
+                Err(1)
+                //~^ ERROR mismatched types
```

Comment:
Hm, I would expect this to suggest `return`. (still better than the status quo ig)